Pipeline triggers (#5483)
pachyderm/pachyderm: doc/docs/master/concepts/advanced-concepts/deferred_processing.md

the conditions is met. To guarantee that they all must be met, add
--trigger-all.

## Embed Triggers in Pipelines

Triggers can also be specified in the pipeline spec and automatically created
when the pipeline is created. For example, this is the edges pipeline from our

> our OpenCV demo modified to only trigger when there is a MB of new images:

Reviewer: a MB ->  1 Megabyte

so we're consistent with prior paragraphs.

Author: ```suggestion
our OpenCV demo modified to only trigger when there is a 1 Megabyte of new images:
```

```
{
  "pipeline": {
    "name": "edges"
  },
  "description": "A pipeline that performs image edge detection by using the OpenCV library.",
  "input": {
    "pfs": {
      "glob": "/*",
      "repo": "images",
      "trigger": {
          "size": "1MB"
      }
    }
  },
  "transform": {
    "cmd": [ "python3", "/edges.py" ],
    "image": "pachyderm/opencv"
  }
}
```

> When you create this pipeline Pachyderm will also create a branch in the input

Reviewer: separate the introductory phrase with a comma:

When you create this pipeline,  Pachyderm will also create a branch in the input

Author: ```suggestion
When you create this pipeline, Pachyderm will also create a branch in the input
```

repo that specifies the trigger and the pipeline will use that branch as its

> input. The name of the branch is auto generated of the form

Reviewer 2: ```suggestion
input. The name of the branch is auto-generated with the form
```

> `<pipeline-name>-trigger-n` you can manually update the heads of these branches

Reviewer 2: ```suggestion
`<pipeline-name>-trigger-n`. You can manually update the heads of these branches
```
